src/teleop/jetson/jetson_teleop.py

```diff
@@ -133,6 +143,39 @@ def ra_mode_callback(self, msg: String) -> None:
         with self._arm_mode_lock:
             self._arm_mode = msg.data
 
+    def brushless_encoder_callback(self, msg: MotorsStatus) -> None:
+        """
+        Callback for brushless encoder values
+        :param msg: MotorsStatus message representing the current status of the brushless motors
+        :return:
+        """
+        with self._joint_state_lock:
+            for i, name in enumerate(msg.name):
+                index_of_joint = self.current_ra_joint_states.name.index(name)
+                self.current_ra_joint_states.position[index_of_joint] = msg.joint_states.position[i]
+                self.current_ra_joint_states.velocity[index_of_joint] = msg.joint_states.velocity[i]
+                self.current_ra_joint_states.effort[index_of_joint] = msg.joint_states.effort[i]
+
+    def brushed_encoder_callback(self, msg: JointState) -> None:
+        """
+        Callback for brushed encoder values
+        :param msg: JointState message representing the current joint states of the brushed motors
+        :return:
+        """
+        with self._joint_state_lock:
+            for i, name in enumerate(msg.name):
+                index_of_joint = self.current_ra_joint_states.name.index(name)
+                self.current_ra_joint_states.position[index_of_joint] = msg.position[i]
+                self.current_ra_joint_states.velocity[index_of_joint] = msg.velocity[i]
+                self.current_ra_joint_states.effort[index_of_joint] = msg.effort[i]
+
+    def publish_joint_states(self) -> None:
+        """
+        Publishes the current joint states to the /joint_states topic
+        :return:
+        """
+        self.joint_state_pub.publish(self.current_ra_joint_states)
```

Comment:
Umm... this needs to be `with self._joint_state_lock` doesn't it?

Reply:
Or is this the case where python just makes things atomic for fun?

Reply:
Ok my knowledge of locks and threading in general is pretty limited but wouldn't you only need the lock when current_ra_joint_states is being set? So that if the publish_joint_states thread is about to publish it will wait until it is done being set by the lock thread? idk I haven't taken 482 yet

Reply:
Imagine you are publishing a variable but something grabs a lock and starts modifying the variable you are using. Because the publish didn't have a lock. Its possible for the variable to change mid-publish

So yes you need a lock to prevent that from happening.

Reply:
^ this. A general rule is that any time there are multiple threads accessing shared data, and at least one thread is modifying that data, then all the threads need to acquire the lock to avoid collisions. Callbacks to rostopics spawn their own threads, so this is multiple threads touching shared data and at least one modifying.